File: Sliding Window/1208. Get Equal Substrings Within Budget/1208.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        costs=[abs(ord(s_v)-ord(t_v)) for s_v,t_v in zip(s,t)]
        res=0
        l=curr=0
        for r in range(len(s)):
            maxCost-=costs[r]
            curr+=1
            if (maxCost>=0):
                res=max(res,curr)
            while (maxCost<0):
                curr-=1
                maxCost+=costs[l]
                l+=1
            
        return res

File: Sliding Window/1208. Get Equal Substrings Within Budget/1208.cpp
#pragma GCC optimize("O3","Oz","unroll-loops","inline-functions","fast-math","march=native","lto","omit-frame-pointer")
class Solution {
public:
    int equalSubstring(string s, string t, int maxCost) {
        ios_base::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);

        int l=0;
        int res=0;
        int curr=0;
        vector<int>count(s.size());
        for (int i=0; i<s.size(); i++) {
            count[i] = abs(s[i] - t[i]);
        }

        for (int r=0; r<count.size(); r++) {
            maxCost-=count[r];
            curr++;
            if (maxCost>=0) {
                res=max(res,curr);
            }
            while (maxCost<0) {
                maxCost+=count[l];
                l++;
                curr--;
            }
        }
        return res;
    }
};